lib/api/multiObjectDelete.js

```diff
@@ -269,6 +269,8 @@ export function getObjMetadataAndDelete(authInfo, canonicalID, request,
             // return DeleteMarkerVersionId equals the versionID of the marker
             // you just generated and DeleteMarker tag equals true
             if (deleteInfo.newDeleteMarker) {
+                // Adding a delete marker should add one to numberOfObjects
```

Comment:
Shouldn't the above TODO be removed now?

Overall I think the whole set of comments here should be clarified and grouped into a single explanation block. I'm not sure why we decrement `numOfObjects`, is it because we're supposed to have incremented it when we created the delete marker first? A more explicit comment would be welcome.